fmriprep/workflows/bold/registration.py

```diff
@@ -479,41 +480,59 @@ def init_bbreg_wf(use_bbr, bold2t1w_dof, bold2t1w_init, omp_nthreads, name='bbre
             LOGGER.warning("Initializing BBR with header; affine fallback disabled")
             use_bbr = True
 
+    # Define both nodes, but only connect conditionally
+    mri_coreg = pe.Node(
+        MRICoregRPT(dof=bold2t1w_dof, sep=[4], ftol=0.0001, linmintol=0.01,
+                    generate_report=not use_bbr),
+        name='mri_coreg', n_procs=omp_nthreads, mem_gb=5)
+
+    bbregister = pe.Node(
+        BBRegisterRPT(
+            dof=bold2t1w_dof,
+            contrast_type='t2',
+            init="header" if bold2t1w_init == "header" else Undefined,
+            registered_file=True,
+            out_lta_file=True,
+            generate_report=True
+        ),
+        name='bbregister', mem_gb=12
+    )
```

Comment:
I find this clearer than having to think about the semantics of explicitly setting `Undefined`, but I recognize it's fine, so I'll leave the option to you.

```suggestion
            registered_file=True,
            out_lta_file=True,
            generate_report=True
        ),
        name='bbregister', mem_gb=12
    )
    if bold2t1w_init == "header":
        bbregister.inputs.init = "header"
```

If you do switch back, don't forget to remove the import.